packages/block-editor/src/components/inspector-controls/fill.js

```diff
@@ -73,15 +70,19 @@ function ToolsPanelInspectorControl( { children, resetAllFilter, fillProps } ) {
 		};
 	}, [ resetAllFilter, registerResetAllFilter, deregisterResetAllFilter ] );
```

Comment:
This effect needs to be rewritten, too. The `fillProps` value used to be `toolsPanelContext`, but now the context has moved into the `forwardedContext` array. The `register`/`deregister` fields are no longer there.

You need to move the registration _under_ the `ToolsPanelContext.Provider`. Into the `innerMarkup` value:
```js
function RegisterResetAll( { resetAllFilter, children } ) {
  const { register, deregister } = useContext( ToolsPanelContext );
  useEffect( () => {
    if ( resetAllFilter && register && deregister ) {
      register( resetAllFilter );
      return () => { deregister( resetAllFilter ); };
    }
  }, [ resetAllFilter, register, deregister ] );
  return children;
};

const innerMarkup = <RegisterResetAll resetAllFilter={ ... }>{ children }</RegisterResetAll>;
```

Reply:
Good point, done in https://github.com/WordPress/gutenberg/pull/50278/commits/8c3c09e5a0007549f93eb5fa84c1e2752ced347d.

Also, cc'ing @aaronrobertshaw to make sure that the registration/deregistration still works as expected!

Reply:
I couldn't see any issues with ToolsPanelItem registration or deregistration in my testing. At a glance, the code changes make sense to me also.